nodejs/awsx/apigateway/authorizer.ts

```diff
@@ -0,0 +1,355 @@
+    uri: pulumi.Input<string> | aws.lambda.Function;
+
+    /**
+     * Credentials required for invoking the authorizer in the form of an ARN of an IAM execution role.
+     * For example, "arn:aws:iam::account-id:IAM_role".
+     */
+    credentials: pulumi.Input<string>;
+}
+
+/** @internal */
+export function isLambdaAuthorizerInfo(info: LambdaAuthorizerInfo | aws.lambda.EventHandler<AuthorizerEvent, AuthorizerResponse>): info is LambdaAuthorizerInfo {
+    return (<LambdaAuthorizerInfo>info).uri !== undefined;
+}
+
+/** @internal */
+export function isLambdaFunction(uri: pulumi.Input<string> | aws.lambda.Function): uri is aws.lambda.Function {
+    return (<aws.lambda.Function>uri).invokeArn !== undefined;
+}
+
+/** @internal */
+export function getIdentitySource(identitySources: string[] | undefined): string {
+    if (identitySources) {
+        return identitySources.join(", ");
+    }
+    return "";
+}
+
+/** @internal */
+export function createRoleWithAuthorizerInvocationPolicy(authorizerName: string, authorizerLambda: aws.lambda.Function): aws.iam.Role {
+    const policy = aws.iam.assumeRolePolicyForPrincipal({ "Service": ["lambda.amazonaws.com", "apigateway.amazonaws.com"] });
+    const role = new aws.iam.Role(authorizerName + "-authorizer-role", {
+        assumeRolePolicy: JSON.stringify(policy),
+    });
+
+    // Add invocation policy to lambda role
+    const invocationPolicy = new aws.iam.RolePolicy(authorizerName + "-invocation-policy", {
+        policy: pulumi.interpolate`{
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Action": "lambda:InvokeFunction",
+                        "Effect": "Allow",
+                        "Resource": "${authorizerLambda.arn}"
+                    }
+                ]
+            }`,
+        role: role.id,
+    });
+    return role;
+}
+
+/**
+ * AuthorizerResponse simplifies creating an AuthorizerResponse.
+ * @param principalId - unique identifier for the user
+ * @param effect - whether to "Allow" or "Deny" the request
+ * @param resource - the API method to be invoked (typically event.methodArn)
+ * @param context - key-value pairs that are passed from the authorizer to the backend Lambda
+ * @param apiKey - if the API uses a usage plan, this must be set to one of the usage plan's API keys
+ */
+export function AuthorizerResponse(principalId: string, effect: Effect, resource: string, context?: AuthResponseContext, apiKey?: string): AuthorizerResponse {
+    const response: AuthorizerResponse = {
+        principalId: principalId,
+        policyDocument: {
+            Version: "2012-10-17",
+            Statement: [{
+                Action: "execute-api:Invoke",
+                Effect: effect,
+                Resource: resource,
+            }],
+        },
+    };
+
+    if (context) {
+        response.context = context;
+    }
+
+    if (apiKey) {
+        response.usageIdentifierKey = apiKey;
+    }
```

Comment:
is it necessary to do the actual `if` checks here? if you don't have a context or key, what's the harm in assigning that to the response?